// rhc-services-library/fesm2015/rhc-services-library.js
import { Injectable, NgModule } from '@angular/core';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CommonApiService {
    /**
     * @param {?} http
     */
    constructor(http) {
        this.http = http;
    }
    /**
     * @param {?} url
     * @return {?}
     */
    commonApiGetService(url) {
        return this.http.get(url);
    }
    /**
     * @param {?} url
     * @param {?} body
     * @return {?}
     */
    commonApiPostService(url, body) {
        return this.http.post(url, body);
    }
}
CommonApiService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
CommonApiService.ctorParameters = () => [
    { type: HttpClient, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class HeaderModule {
}
HeaderModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HttpClientModule
                ],
                declarations: [],
                providers: [CommonApiService],
                exports: []
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { HeaderModule, CommonApiService as ɵa };
