// rhc-services-library/fesm5/rhc-services-library.js
import { Injectable, NgModule } from '@angular/core';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var CommonApiService = /** @class */ (function () {
    function CommonApiService(http) {
        this.http = http;
    }
    /**
     * @param {?} url
     * @return {?}
     */
    CommonApiService.prototype.commonApiGetService = /**
     * @param {?} url
     * @return {?}
     */
    function (url) {
        return this.http.get(url);
    };
    /**
     * @param {?} url
     * @param {?} body
     * @return {?}
     */
    CommonApiService.prototype.commonApiPostService = /**
     * @param {?} url
     * @param {?} body
     * @return {?}
     */
    function (url, body) {
        return this.http.post(url, body);
    };
    CommonApiService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    CommonApiService.ctorParameters = function () { return [
        { type: HttpClient, },
    ]; };
    return CommonApiService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HeaderModule = /** @class */ (function () {
    function HeaderModule() {
    }
    HeaderModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        HttpClientModule
                    ],
                    declarations: [],
                    providers: [CommonApiService],
                    exports: []
                },] },
    ];
    return HeaderModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { HeaderModule, CommonApiService as ɵa };
