// rhc-services-library/bundles/rhc-services-library.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('rhc-services-library', ['exports', '@angular/core', '@angular/common/http', '@angular/common'], factory) :
    (factory((global['rhc-services-library'] = {}),global.ng.core,global.ng.common.http,global.ng.common));
}(this, (function (exports,core,http,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var CommonApiService = (function () {
        function CommonApiService(http$$1) {
            this.http = http$$1;
        }
        /**
         * @param {?} url
         * @return {?}
         */
        CommonApiService.prototype.commonApiGetService = /**
         * @param {?} url
         * @return {?}
         */
            function (url) {
                return this.http.get(url);
            };
        /**
         * @param {?} url
         * @param {?} body
         * @return {?}
         */
        CommonApiService.prototype.commonApiPostService = /**
         * @param {?} url
         * @param {?} body
         * @return {?}
         */
            function (url, body) {
                return this.http.post(url, body);
            };
        CommonApiService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        CommonApiService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
            ];
        };
        return CommonApiService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var HeaderModule = (function () {
        function HeaderModule() {
        }
        HeaderModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            http.HttpClientModule
                        ],
                        declarations: [],
                        providers: [CommonApiService],
                        exports: []
                    },] },
        ];
        return HeaderModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.HeaderModule = HeaderModule;
    exports.ɵa = CommonApiService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
